text/0000-cargo-report-future-incompat.md

```diff
@@ -0,0 +1,720 @@
+   = note: fields of packed structs might be misaligned: dereferencing a misaligned pointer or even just creating a misaligned reference is undefined behavior
+   Compiling unwary v0.1.0 (/tmp/unwary)
+    Finished dev [unoptimized + debuginfo] target(s) in 0.30s
+unwary % touch src/main.rs
+unwary % cargo build                                                # ... but no warning issued about still-present problem in path dependency.
+   Compiling unwary v0.1.0 (/tmp/unwary)
+    Finished dev [unoptimized + debuginfo] target(s) in 0.30s
+unwary %
+```
+
+Therefore, if the developer does not do something to address the
+warning the first time they see it, it is likely to go unaddressed
+until the developer upgrades to a version of Rust that actually fails
+to compile `brash`. This is a frustrating developer experience.
+
+Cargo passes `--cap-lints=allow` on upstream dependencies for good
+reason, as discussed in [Rust RFC 1193][] and the comment thread from
+[rust-lang/rust#59658][].
+For cases like future-incompatibility lints, which
+are more severe warnings for the long-term viability of a crate, we
+need to provide some feedback to the `unwary` maintainer. 
+
+But this feedback should not be *just* the raw diagnostic output for
+`brash`! The developer of a crate like `unwary` typically cannot do
+anything in the short term about warnings emitted by upstream crates,
+
+ * (The `unwary` developer *can* file issues against `brash`a or even
+    contribute code to fix `brash`, but that does not resolve the
+    immediate problem until new version of `brash` is released by its
+    maintainer.)
+
+Therefore the diagnostics associated with building upstream
+`brash` are usually just noise from the viewpoint of `unwary`'s
+maintainer.
+
+[Rust RFC 1193]: https://github.com/rust-lang/rfcs/blob/master/text/1193-cap-lints.md
+
+[rust-lang/rfcs#1193]: https://github.com/rust-lang/rfcs/pull/1193
+
+[rust-lang/rust#59658]: https://github.com/rust-lang/rust/issues/59658
+
+[rust-lang/rust#27260]: https://github.com/rust-lang/rust/pull/27260
+
+[rust-lang/cargo#1830]: https://github.com/rust-lang/cargo/pull/1830
+
+Therefore, we want to continue passing `--cap-lints=allow` for
+upstream dependencies. But we also want `rustc` to tell `cargo` (via
+some channel) about when future-incompatibility lints are triggered,
+and we want `cargo` to provide a succinct report of the triggers.
+
+This RFC suggests the provided feedback take the form of a summary at
+the end of cargo's build of `unwary`, as illustrated in the explanation below.
+
+Furthermore, we want the feedback to provide guidance as to how the
+`unwary` maintainer can address the issue. Here are some potential
+forms this additional guidance could take.
+
+ * If cargo is not running in "offline mode", it can take the
+   future-incompatibilty signaling as an opportunity to query
+   `crates.io` to find out if a newer version of the upstream crate is
```

Comment:
If the local index gets updated with the available versions (even if the local system has not updated to those latest versions on any of its crates' dependencies), then this sounds fine. I just want to clearly convey the idea that a version query *can* happen; the details of its form can be left to the cargo team to fine-tune. I'll try to revise this text accordingly.

Reply:
I changed the language here to follow @ehuss 's suggestion in 5d4d98ca301aac5d41b22010985b306ecb1dae2a ; but since I am not 100% sure that the central index is updated with the necessary metadata often enough, I kept in an aside about querying crates.io as well.